src/System.IO.FileSystem.AccessControl/src/System/Security/AccessControl/FileSecurity.cs

```diff
@@ -489,7 +489,7 @@ public bool RemoveAccessRule(FileSystemAccessRule rule)
                 FileSystemAccessRule fsrule = rules[i] as FileSystemAccessRule;
 
                 if ((fsrule != null) && (fsrule.FileSystemRights == rule.FileSystemRights)
-                    && (fsrule.IdentityReference == rule.IdentityReference)
+                    && (fsrule.IdentityReference.Translate(typeof(SecurityIdentifier)) == rule.IdentityReference.Translate(typeof(SecurityIdentifier)))
```

Comment:
Why are product changes necessary?

Reply:
This issue is not with tests but with API implementation more. Anyway I will remove changes here and I will change tests but not API implementation. Sorry.